modules/logging/src/main/java/io/oasp/module/logging/common/impl/SecureLogging.java

```diff
@@ -0,0 +1,242 @@
+   */
+  private static void initMarkers() {
+
+    if (initialized)
+      return;
+
+    Class<?> cExtClass = hasExtClass(extClass);
+
+    if (cExtClass.isAssignableFrom(String.class)) {
+      createDefaultMarkers();
+    } else {
+      createMultiMarkers(cExtClass);
+    }
+
+    if (!initialized)
+      LOG.warn("SecureLogging Markers could not be initialized!");
+    else
+      LOG.debug("SecureLogging Markers created: '{}', ...", markerSecurSuccConfid.getName());
+    return;
+  }
+
+  private static void createDefaultMarkers() {
+
+    LOG.debug("Creating default markers.");
+    markerSecurSuccConfid = MarkerFactory.getDetachedMarker(SECURITY_SUCCESS_CONFIDENTIAL_MARKER_NAME);
+    markerSecurFailConfid = MarkerFactory.getDetachedMarker(SECURITY_FAILURE_CONFIDENTIAL_MARKER_NAME);
+    markerSecurAuditConfid = MarkerFactory.getDetachedMarker(SECURITY_AUDIT_CONFIDENTIAL_MARKER_NAME);
+    initialized = true;
+  }
+
+  private static void createMultiMarkers(Class<?> cExtClass) {
+
+    LOG.debug("Creating MultiMarkers.");
+
+    Object objExtClass = null;
+    try {
+      objExtClass = cExtClass.newInstance();
+      Class<?>[] paramTypes = { Marker[].class }; // the method to invoke is "getMarker(Marker... markers)".
+      Method method = cExtClass.getMethod(methodName, paramTypes);
+
+      Marker[] markerArray = { MarkerFactory.getDetachedMarker(SECURITY_SUCCESS_MARKER_NAME),
+      MarkerFactory.getDetachedMarker(CONFIDENTIAL_MARKER_NAME) };
+      markerSecurSuccConfid = (Marker) method.invoke(objExtClass, (Object) markerArray);
+      markerArray[0] = MarkerFactory.getDetachedMarker(SECURITY_FAILURE_MARKER_NAME);
+      markerSecurFailConfid = (Marker) method.invoke(objExtClass, (Object) markerArray);
+      markerArray[0] = MarkerFactory.getDetachedMarker(SECURITY_AUDIT_MARKER_NAME);
+      markerSecurAuditConfid = (Marker) method.invoke(objExtClass, (Object) markerArray);
+      initialized = true;
+
+    } catch (Exception e) {
+      LOG.warn("Error getting Method '{}' of Class '{}'. Falling back to default.", methodName, cExtClass.getName());
+      LOG.warn("Exception occurred.", e);
+      e.printStackTrace();
+      createDefaultMarkers();
+    }
+  }
+
+  /**
+   * @return The given {@link Class} if parameter 'className' can be resolved, otherwise {@link String}.class.
+   */
+  private static Class<?> hasExtClass(String className) {
+
+    Class<?> cExtClass;
+    try {
+      cExtClass = Class.forName(className);
+      return cExtClass;
+    } catch (Exception e) {
+      LOG.debug("Class '{}' or one of its dependencies is not present.", className);
+      cExtClass = String.class;
+      return cExtClass;
+    }
+  }
+
+  /**
+   * @return extClass
+   */
+  public static String getExtClass() {
+
+    return extClass;
+  }
+
+  /**
+   * @param extClass new value of {@link #getExtClass}.
+   */
+  public static void setExtClass(String extClass) {
```

Comment:
Seems over engineered: getting and setting of "extClass" and "methodName" is IMHO a flexibility that none will actually need. Even worse I assume that it might introduce bugs with bootstrapping when the setter is called too late in the startup process. In the end the arguments of the getMarker method are also hard-coded. For KISS I would propose to just change these two attributes to constants and remove these getter/setter pairs.